src/rgw/CMakeLists.txt

```diff
@@ -33,6 +33,21 @@ endfunction()
 
 find_package(ICU 52.0 COMPONENTS uc REQUIRED)
 
+find_package(PkgConfig QUIET)
+if(PkgConfig_FOUND)
+  pkg_check_modules(LMDB QUIET lmdb)
+endif()
+
+if(NOT LMDB_FOUND)
+  find_path(LMDB_INCLUDE_DIR NAMES lmdb.h)
+  find_library(LMDB_LIBRARIES NAMES lmdb)
+  if(LMDB_INCLUDE_DIR AND LMDB_LIBRARIES)
+    set(LMDB_FOUND TRUE)
+  else()
+    message(FATAL_ERROR "LMDB not found. Please install liblmdb-dev or lmdb-devel")
+  endif()
+endif()
```

Comment:
cc @mattbenjamin @dang

we already have lmdb code copied into `src/rgw/driver/posix/lmdb-safe*`. is "lmdb-safe" the same thing we're using here? if so, can we please find a way for both of these features to share a single dependency? ideally that would come from system packages if they're available